msg-socket/src/req/socket.rs

```diff
@@ -250,8 +262,8 @@ mod tests {
         let request = Bytes::from_static(b"test request");
         let mut futures = Vec::new();
     
-        // Sending an arbitrary large amount of requests
-        for _ in 0..20 {
+        // Sending one more request than max
+        for _ in 0..=max_pending_requests {
```

Comment:
This test should fail, but it passes. One way to make sure it fails is to use a `RepSocket` that takes a long time to reply (sleep for 100ms), and gradually send more requests (each with a short sleep between them)